static/app/utils/discover/genericDiscoverQuery.tsx

```diff
@@ -138,10 +143,15 @@ class _GenericDiscoverQuery<T, P> extends React.Component<Props<T, P>, State<T>>
   }
 
   getPayload(props: Props<T, P>) {
-    if (this.props.getRequestPayload) {
-      return this.props.getRequestPayload(props);
+    const payload = this.props.getRequestPayload
+      ? this.props.getRequestPayload(props)
+      : props.eventView.getEventsAPIPayload(props.location);
+
+    if (this.props.disablePagination) {
+      delete payload.cursor;
```

Comment:
I'm guessing you're doing this because the cursor is shared on the page and causing both requests to have `page 2` after you click next? This work for now, but long term I think we might consider a different solution since we want something like independent pagination for widgets on a page anyway. In general I'd say having something like cursor be explicit and set to null is better than having 2 props that modify pagination.

Reply:
I opted to specify an explicit cursor of `0:0:1` and enabled `noPagination`. This combined has the effect of remaining on the first page and prevents the same request from being fired.